### ISSUE TITLE
indicate next edit suggestion availability to screen reader users

### ISSUE THREAD

**Issue by meganrogge** [microsoft/vscode#230322]
1. Enable a screen reader and the settings to have edit suggestions
2. Make an edit - here, I accepted an inline completion (thus the view this in the accessible view message)
3. Visually, I see an edit suggestion. There's no indication of this to screen reader users. We probably should have an accessibility signal for this and want to indicate how to navigate there via keyboard in the editor's a11y help dialog.

![Image](https://github.com/user-attachments/assets/46434549-aa36-401e-87c2-e240b76e7ff5)

**Comment by AFre100** [comment]
@meganrogge good call! I return in December, so should be able to have this in December/Jan

**Comment by AFre100** [comment]
@meganrogge Since my leave is going to last a few weeks later than expected, let's update the ETA to early Feb

**Comment by AFre100** [comment]
Hi @rperez030 @jooyoungseo  , wanted to ask what kind of audio feedback/flow might be most helpful for screen reader users in this scenario — ARIA (voice) and/or Audio Cue? Thanks!

**Comment by meganrogge** [comment]
We will have both here, so users can decide which they'd prefer.

**Comment by AFre100** [comment]
@meganrogge et all, There are a few potential moments for audio feedback: 

Notifying users when 1) the edit suggestion menu first pops up 2) there's an *additional* edit suggestion available in the menu user is already inside of 3) when an edit suggestion has been accepted/triggered

Which are you targeting, and if it's multiple, how many sounds are you thinking?

**Comment by meganrogge** [comment]
Probably a sound for when it's popped up and a sound for when it's applied.

**Comment by AFre100** [comment]
@rperez030 @jooyoungseo Also wanted to ask if there were any specific expectations in the screenreader user world, in terms of a pattern for "edit suggestion availability" or

**Comment by AFre100** [comment]
@meganrogge and all, would the sound for "edit suggestion applied" simply be whatever will be used for "[Code Action Request applied](https://github.com/microsoft/vscode/issues/233765)" ? They are effectively the same, right? 

(Meaning, only one new sound would need to be developed for when menu first pops up)

> Probably a sound for when it's popped up and a sound for when it's applied.

**Comment by meganrogge** [comment]
My answer: No, as usual, we need unique sounds so it's not confusing for users. 

Let's see what @jooyoungseo and @rperez030 think though.

**Comment by AFre100** [comment]
@meganrogge Thanks -- I wrongly thought the two concepts ("Code Action Request applied" and "edit suggestion applied" ) were two ways to describe one concept. What's the defining difference between the two?

**Comment by accesswatch** [comment]
I am a screen reader user myself and I do agree that unique sounds for clarity are best here.

I do think some adjustments on accessibility signals for the details should be considered but I will leave that for another day 😊.

Jeff Bishop


From: Megan Rogge ***@***.***>
Sent: Tuesday, January 21, 2025 2:59 PM
To: microsoft/vscode ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [EXT] Re: [microsoft/vscode] indicate next edit suggestion availability to screen reader users (Issue #230322)


External Email

________________________________
My answer: No, as usual, we need unique sounds so it's not confusing for users.

Let's see what @jooyoungseo<https://github.com/jooyoungseo> and @rperez030<https://github.com/rperez030> think though.

**Comment by AFre100** [comment]
> I am a screen reader user myself and I do agree that unique sounds for clarity are best here.
> 
> I do think some adjustments on accessibility signals for the details should be considered but I will leave that for another day 😊.
> 
> Jeff Bishop
> 
> 
> From: Megan Rogge ***@***.***>
> Sent: Tuesday, January 21, 2025 2:59 PM
> To: microsoft/vscode ***@***.***>
> Cc: Subscribed ***@***.***>
> Subject: [EXT] Re: [microsoft/vscode] indicate next edit suggestion availability to screen reader users (Issue [#230322](https://github.com/microsoft/vscode/issues/230322))
> 
> 
> External Email
> […](#)

@accesswatch Hello, and thank you for the feedback! I just DM'd you on facebook re adjustments :)

**Comment by AFre100** [comment]
@meganrogge et all, does this flow sound right, and if so, where do you foresee aria alerts w/in it? 

-edit suggestion menu first pops up [New Edit Suggestion Alert Sound #1] 
-User navigates through menu of 1+ edit suggestion(s) and triggers one[No additional sounds] 
-An edit suggestion has been applied [New Edit Suggestion Applied Sound #2]

**Comment by meganrogge** [comment]
Yep!

**Comment by meganrogge** [comment]
@AFre100, what is the ETA for this sound?

**Comment by AFre100** [comment]
@meganrogge  if the sound file needs to be ready by mid-April for the April Insiders (with the 17th as the monthly deadline), I can deliver it by 4/14, which would be ideal.  Alternatively, if late March/early April delivery adds significant value, I can commit to that.   How would you like to proceed?

**Comment by hediet** [comment]
Duplicate of https://github.com/microsoft/vscode/issues/228033, closing in favor of https://github.com/microsoft/vscode/issues/243772

We now play the same sound for inline completions and inline edits. Both present the suggestion in the accessible help view.
Adopting a new sound would be nice!

**Comment by meganrogge** [comment]
@hediet thank you for working on this. It is very confusing to reuse sounds for screen reader users. However, there's a tradeoff in that the feature is not accessible at all while we wait for a new sound. cc @AFre100

**Comment by AFre100** [comment]
@meganrogge I just realized I might have unnecessarily frontloaded the suggested deadline for April Insiders (I said 4/14). What’s the latest date and time we can set for the deadline?

**Comment by meganrogge** [comment]
@AFre100 April 17th at 4pm CST?

**Comment by AFre100** [comment]
> [@meganrogge](https://github.com/meganrogge) et all, does this flow sound right, and if so, where do you foresee aria alerts w/in it?
> 
> -edit suggestion menu first pops up [New Edit Suggestion Alert Sound 1] -User navigates through menu of 1+ edit suggestion(s) and triggers one�[No additional sounds] -An edit suggestion has been applied [New Edit Suggestion Applied Sound 2]

@meganrogge 4/17 4pm CST is great. Can you please help make sure I'm on the same page as to how the flow works?: 
Am I correct in flagging that this flow will still require 2 new sounds as we discussed, here? I ask because I recently delivered a "Kept" sound -- that Copilot sound is totally separate from this "Accept" sound, correct? And if so, should “Applied Sound 2” actually be called "Accept Sound" or something similar?

**Comment by meganrogge** [comment]
Hi @AFre100 , 

`Kept` is happening when a user decides to keep a change that happened using the Copilot chat edits flow.

`Accept` will occur when an inline next edit suggetion is reviewed and approved by the user (see gif above).

I am not sure if reuse is okay here - we'll want feedback from @jooyoungseo and @rperez030 . 

The sound we definitely do need is the one that indicates the suggestion has appeared.

**Comment by meganrogge** [comment]
Verified that the sound plays, but found that a user cannot inspect and accept the edit https://github.com/microsoft/vscode/issues/244651
